### PR TITLE
More removing of timeouts.

### LINKED ISSUE #33931: Remove test timeouts by default

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/33931 and https://github.com/flutter/flutter/issues/18507.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @tvolkert

**Comment by Hixie** [comment]
https://groups.google.com/forum/#!topic/flutter-announce/pEzegN6upf4

File: packages/flutter_test/lib/src/matchers.dart [hunk 1]
```diff
@@ -1646,28 +1646,23 @@ class _MatchesReferenceImage extends AsyncMatcher {
...
     return binding.runAsync<String>(() async {
       final ui.Image image = await imageFuture;
-      final ByteData bytes = await image.toByteData()
-        .timeout(const Duration(seconds: 10), onTimeout: () => null);
-      if (bytes == null) {
-        return 'Failed to generate an image from engine within the 10,000ms timeout.';
-      }
+      final ByteData bytes = await image.toByteData();
+      if (bytes == null)
+        return 'could not be encoded.';
```
Comment (tvolkert) [inline packages/flutter_test/lib/src/matchers.dart]: Nit: "Image could not be encoded"

Comment (tvolkert) [reply packages/flutter_test/lib/src/matchers.dart]: Oh n/m - this is going to be appended to the matcher's failure reason, preceded by a "which" isn't it?

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_test/lib/src/matchers.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_test/lib/src/matchers.dart]: I'm guessing yes? The code was inconsistent before, I made it all use this style.

**Comment by Hixie** [comment]
Based on discussion with @dnfield in https://github.com/flutter/flutter/issues/33931 I'm going to put back the backstop timeout.

**Comment by Hixie** [comment]
I negotiated myself to 10 minutes being the default timeout.

Attempted to update the API docs and so forth. PTAL.

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**